lib/request-handler.js

```diff
@@ -143,19 +142,19 @@ RequestHandler.prototype.log = utils.log;
 /**
  * Gets an available connection and sends the request
  * @param request
- * @param {QueryOptions} options
+ * @param {QueryOptions|null} options
  * @param {Function} callback
  */
 RequestHandler.prototype.send = function (request, options, callback) {
   if (this.request === null) {
-    //Set the first time
-    if (options && options.captureStackTrace) {
+    options = options || utils.emptyObject;
+    if (options.captureStackTrace) {
       //noinspection JSUnresolvedFunction
       Error.captureStackTrace(this.stackContainer);
     }
-    if (options && options.retry) {
-      this.retryPolicy = options.retry;
-    }
+    var profile = options.executionProfile || this.profileManager.getDefault();
```

Comment:
Also technically couldn't:

```javascript
this.retryPolicy = options.retry || profile.retry;
```

simply be:

```javascript
this.retryPolicy = options.retry;
```

Since options.retry should be set to profile.retry if there is no query level retry in createQueryOptions?